app/Http/Controllers/Api/VolumeBackupsController.php

```
@@ -0,0 +1,415 @@
+        $created = $backup->wasRecentlyCreated;
+
+        auditLog('api.volume_backup.schedule_set', [
+            'team_id' => $teamId,
+            'resource_type' => $resourceType,
+            'resource_uuid' => $resource->uuid,
+            'storage_uuid' => $storage->uuid,
+            'backup_uuid' => $backup->uuid,
+        ]);
+
+        return response()->json($this->responseData($backup, $storage, $s3Storage, $created), $created ? 201 : 200);
+    }
```

PHPMD (2.15.0)

[warning] The method upsert() has a Cyclomatic Complexity of 18. The configured cyclomatic complexity threshold is 10. (undefined)

(CyclomaticComplexity)